### ISSUE TITLE
Experiment with using react server side

### ISSUE THREAD

**Issue by designfrontier** [ansble/monument#294]
Using the config object changes and the render changes from #292 #291 run a site using react and see if there are things that need to be changed.

Also look at whether autodetection of the engine and if there are defaults that make sense if we can autodetect

**Comment by barrotsteindev** [comment]
Hello,
I would like to give this a shot.

**Comment by nitin42** [comment]
I consider this to be the case of server-side rendering with React using the configObject. If so then I will start experimenting and will let you know about this.

**Comment by designfrontier** [comment]
@nitin42 yeah that is pretty much what it is. You'll probably want to base your work on https://github.com/ansble/monument/pull/308 which is still in progress... Let me know if you have questions as you go along

**Comment by TheLordHighExecutioner** [comment]
New guy here. What do you mean by using React "server-side"?

**Comment by designfrontier** [comment]
so using React for server side templating to allow for first render happening on the server. Think Progressive Enhancement or Isomorphic JS.

**Comment by kritisingh1** [comment]
@designfrontier I would like to work on this issue. Please let me know if I may. This would be my first contribution to open source

**Comment by designfrontier** [comment]
Awesome! Go for it
